presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -424,10 +424,10 @@ private static String quotedName(String name)
         return '"' + name.replace("\"", "\"\"") + '"';
     }
 
-    public static TableScan getTableScan(TupleDomain<IcebergColumnHandle> predicates, Optional<Long> snapshotId, Table icebergTable)
+    public static TableScan getTableScan(TupleDomain<IcebergColumnHandle> predicates, Optional<Long> snapshotId, Table icebergTable, ConnectorSession session)
     {
         Expression expression = ExpressionConverter.toIcebergExpression(predicates);
-        TableScan tableScan = icebergTable.newScan().filter(expression);
+        TableScan tableScan = icebergTable.newScan().metricsReporter(new RuntimeStatsMetricsReporter(session.getRuntimeStats())).filter(expression);
```

Comment:
Rather than passing down the session which calls `getRuntimeStats`, let's make this take `RuntimeStats` directly

Rather than passing down the session which calls `getRuntimeStats`, let's make this take `RuntimeStats` directly
```suggestion
    public static TableScan getTableScan(TupleDomain<IcebergColumnHandle> predicates, Optional<Long> snapshotId, Table icebergTable, RuntimeStats runtimeStats)
    {
    Expression expression = ExpressionConverter.toIcebergExpression(predicates);
    TableScan tableScan = icebergTable.newScan().metricsReporter(new RuntimeStatsMetricsReporter(runtimeStats)).filter(expression);
```